src/python/pants/backend/go/util_rules/analyze_package.go

```diff
@@ -60,9 +65,12 @@ type Package struct {
 	//	//go:embed a* b.c
 	// then the list will contain those two strings as separate entries.
 	// (See package embed for more details about //go:embed.)
-	EmbedPatterns      []string `json:",omitempty"` // patterns from GoFiles, CgoFiles
-	TestEmbedPatterns  []string `json:",omitempty"` // patterns from TestGoFiles
-	XTestEmbedPatterns []string `json:",omitempty"` // patterns from XTestGoFiles
+	EmbedPatterns      []string  `json:",omitempty"` // patterns from GoFiles, CgoFiles
+	EmbedConfig        *EmbedCfg `json:",omitempty"` // files matching the EmbedPatterns
```

Comment:
Note: The `go list` command returns lists of files to embed here and relies on the invoking tool to apply wildcards. That approach won't work in Python since we want to use the glob algorithm in the Go standard library which seems to differ somewhat from the one in Python (around certain syntax). 

`go list` returns `EmbedFiles`, `TestEmbedFiles`, and `XTestEmbedFiles`. Since we analyze the wildcards in Go, this analysis just returns the exact embedcfg to use.

Reply:
This would probably be good to put in a comment. Wdyt?

Reply:
As discussed in chat, will do in follow-up PR.